python/ray/data/_internal/datasource/iceberg_datasource.py

```diff
@@ -113,6 +124,9 @@ def __init__(
             catalog_kwargs: Optional arguments to use when setting up the Iceberg
                 catalog
         """
+        # Initialize parent class to set up predicate pushdown mixin
+        super().__init__()
+
```

Comment:
Out-of-scope for this PR since this is the existing behavior, but I think the requirement to call `super().__init__()` for predicate pushdown to work is really implicit.

Like, if you're extending a datasource and want predicate pushdown to work, I have no clue how you'd know to do this

I think we should avoid this sort of implicitness when we introduce DatasourceV2

Reply:
Just make __init__ an abstract method in the mixin